### PR TITLE
80286 Alter attachment table

### PR DESCRIPTION
**Pull request by bratli**
DevOps#80286

### CODE REVIEW

File: src/Equinor.ProCoSys.IPO.Command/InvitationCommands/DeleteAttachment/DeleteAttachmentCommandHandler.cs [hunk 1]
```diff
@@ -42,6 +42,7 @@ public async Task<Result<Unit>> Handle(DeleteAttachmentCommand request, Cancella
             await _blobStorage.DeleteAsync(fullBlobPath, cancellationToken);
 
             invitation.RemoveAttachment(attachment);
```
Comment (janid1967) [inline src/Equinor.ProCoSys.IPO.Command/InvitationCommands/DeleteAttachment/DeleteAttachmentCommandHandler.cs:44]: Add tests in DeleteAttachmentCommandHandlerTests, asserting that invitation got 1 less attachment

File: src/Equinor.ProCoSys.IPO.Command/InvitationCommands/DeleteAttachment/DeleteAttachmentCommandHandler.cs [hunk 2]
```diff
@@ -42,6 +42,7 @@ public async Task<Result<Unit>> Handle(DeleteAttachmentCommand request, Cancella
...
+            _invitationRepository.RemoveAttachment(attachment);
```
Comment (janid1967) [inline src/Equinor.ProCoSys.IPO.Command/InvitationCommands/DeleteAttachment/DeleteAttachmentCommandHandler.cs:45]: Add tests in DeleteAttachmentCommandHandlerTests, asserting that attachment table got 1 less attachment

File: src/Equinor.ProCoSys.IPO.Domain/AggregateModels/InvitationAggregate/IInvitationRepository.cs [hunk 3]
```diff
@@ -8,5 +8,6 @@ public interface IInvitationRepository : IRepository<Invitation>
         void RemoveCommPkg(CommPkg commPkg);
         void RemoveMcPkg(McPkg mcPkg);
         void RemoveParticipant(Participant participant);
+        void RemoveAttachment(Attachment attachment);
```
Comment (janid1967) [inline src/Equinor.ProCoSys.IPO.Domain/AggregateModels/InvitationAggregate/IInvitationRepository.cs:11]: Missing test for RemoveAttachment in InvitationRepositoryTests.
When investigating InvitationRepositoryTests, I see a lot of test missing there + a lot of existing tests not belonging there.

File: src/Equinor.ProCoSys.IPO.Infrastructure/Repositories/InvitationRepository.cs [hunk 4]
```diff
@@ -45,5 +45,8 @@ public void RemoveMcPkg(McPkg mcPkg)
 
         public void RemoveParticipant(Participant participant)
             => _context.Participants.Remove(participant);
+
+        public void RemoveAttachment(Attachment attachment)
```
Comment (janid1967) [inline src/Equinor.ProCoSys.IPO.Infrastructure/Repositories/InvitationRepository.cs:49]: Missing test for RemoveAttachment in InvitationRepositoryTests.
When investigating InvitationRepositoryTests, I see a lot of test missing there + a lot of existing tests not belonging there.

File: src/Equinor.ProCoSys.IPO.WebApi/Controllers/Invitation/InvitationsController.cs [hunk 5]
```diff
@@ -206,7 +206,7 @@ public class InvitationsController : ControllerBase
 
         [Authorize(Roles = Permissions.IPO_READ)]
         [HttpGet("{id}/Attachments/{attachmentId}")]
-        public async Task<ActionResult<AttachmentDto>> GetAttachment(
+        public async Task<ActionResult<Query.GetAttachmentById.AttachmentDto>> GetAttachment(
```
Comment (janid1967) [inline src/Equinor.ProCoSys.IPO.WebApi/Controllers/Invitation/InvitationsController.cs:209]: Undo line 209 + 223. Existing AttachmentDto in current namespace seems to be unused. Delete it

File: src/Equinor.ProCoSys.IPO.Infrastructure/Repositories/InvitationRepository.cs (hunk 4 above)
Comment (janid1967) [reply src/Equinor.ProCoSys.IPO.Infrastructure/Repositories/InvitationRepository.cs:49]: ... call me if you don't see test not belonging there

Comment (bratli) [reply src/Equinor.ProCoSys.IPO.Infrastructure/Repositories/InvitationRepository.cs:49]: I have a test, RemoveAttachment_KnownAttachment_ShouldRemoveAttachment, in InvitationRepositoryTests. Should I add more?

File: src/tests/Equinor.ProCoSys.IPO.Command.Tests/InvitationCommands/DeleteAttachment/DeleteAttachmentCommandHandlerTests.cs [hunk 6]
```diff
@@ -17,11 +18,13 @@ public class DeleteAttachmentCommandHandlerTests
         private const string PLANT = "PCS$TESTPLANT";
 
         private Invitation _invitation;
+        private Attachment _attachmentMock;
```
Comment (janid1967) [inline src/tests/Equinor.ProCoSys.IPO.Command.Tests/InvitationCommands/DeleteAttachment/DeleteAttachmentCommandHandlerTests.cs:21]: _attachment

File: src/tests/Equinor.ProCoSys.IPO.Command.Tests/InvitationCommands/DeleteAttachment/DeleteAttachmentCommandHandlerTests.cs [hunk 7]
```diff
@@ -61,23 +66,54 @@ public void Setup()
...
+        }
+
         private class TestableAttachment : Attachment
         {
             public TestableAttachment(string plant, string fileName)
                 : base(plant, fileName)
             {
             }
-
             public void SetId(int id) => Id = id;
```
Comment (janid1967) [inline src/tests/Equinor.ProCoSys.IPO.Command.Tests/InvitationCommands/DeleteAttachment/DeleteAttachmentCommandHandlerTests.cs:117]: remove

File: src/tests/Equinor.ProCoSys.IPO.Infrastructure.Tests/Repositories/InvitationRepositoryTests.cs [hunk 8]
```diff
@@ -21,6 +21,10 @@ public class InvitationRepositoryTests : RepositoryTestBase
         private const int ParticipantId = 1;
         private List<Invitation> _invitations;
         private Mock<DbSet<Invitation>> _dbSetMock;
```
Comment (janid1967) [inline src/tests/Equinor.ProCoSys.IPO.Infrastructure.Tests/Repositories/InvitationRepositoryTests.cs:23]: _invitationSetMock

**Review by janid1967 — APPROVED** [review]

**Pull request merged**